libosmscout-import/src/osmscout/import/GenLocationIndex.cpp

```diff
@@ -94,8 +95,26 @@ namespace osmscout {
     return false;
   }
 
-  bool LocationIndexGenerator::Region::CouldContain(const Region& region) const
+  bool LocationIndexGenerator::Region::CouldContain(const Region& region, bool strict) const
   {
+    if (strict) {
+      // test if the region can be fit into *this when the boundingBox
+      // is taken into account
+      double area_region = region.boundingBox.GetSize(); // reference area
+
+      GeoCoord cornerMin( std::max( region.boundingBox.GetMinLat(), this->boundingBox.GetMinLat() ),
+			  std::max( region.boundingBox.GetMinLon(), this->boundingBox.GetMinLon() ) );
+      
+      GeoCoord cornerMax( std::min( region.boundingBox.GetMaxLat(), this->boundingBox.GetMaxLat() ),
+			  std::min( region.boundingBox.GetMaxLon(), this->boundingBox.GetMaxLon() ) );
+
+      GeoBox bx(cornerMin, cornerMax);
```

Comment:
Actually, since I am correcting formatting (tabs -> spaces), maybe its easier if I push it in. Otherwise we'll have lots of conflicts...